test/test_runner.go

```diff
@@ -581,14 +584,14 @@ func (r *TestRunner) interpreterContractValueHandler(
 
 			switch location := compositeType.Location.(type) {
 			case common.AddressLocation:
-				storage := runtime.NewStorage(
+				// All contracts are deployed on EmulatorBackend's
+				// blockchain, so we construct a storage based on
+				// its ledger.
+				blockchainStorage := runtime.NewStorage(
```

Comment:
Is it maybe possible to just use the same storage, instead of creating a new one and copying over the slabs?

Reply:
I guess it has to do with the fact that the special `setup()` function is the one that makes the deployments.

Reply:
> I'll see if there is an easy way to re-use the storage.

If this proves to require more elaborate work, could we consider leaving this as a technical debt? Releasing this will provide better DevEx :pray: 

```cadence
let randomSource = scriptResult.returnValue! as! RandomBeaconHistory.RandomSource
let expectedRandomSource = RandomBeaconHistory.RandomSource(
    blockHeight: getCurrentBlockHeight() - 1,
    value: [0, 1, 1, 2, 3, 5, 8]
)
Test.assertEqual(expectedRandomSource, randomSource)
```
The above use of `RandomBeaconHistory.RandomSource(...)` would trigger the error in the PR's description.

Reply:
@m-Peter What problems are you running into with the original code?

Reply:
In certain cases, when creating values of types defined in imported contracts, e.g.

```cadence
import Test
import "RandomBeaconHistory"

// ... contracts deployment & setup

let randomSource = scriptResult.returnValue! as! RandomBeaconHistory.RandomSource
let expectedRandomSource = RandomBeaconHistory.RandomSource(
    blockHeight: getCurrentBlockHeight() - 1,
    value: [0, 1, 1, 2, 3, 5, 8]
)
Test.assertEqual(expectedRandomSource, randomSource)
```

we are no longer able to call some methods on the `Test` contract. I get the error:
```shell
error: internal error: slab (0x0.7) not found: slab not found for stored value
```
This comes from this line: https://github.com/onflow/cadence-tools/blob/master/test/test_runner.go#L588-L590
We inject a different storage instance in the `inter`, which does not contain the slabs allocated for the `Test` contract.

Reply:
Just committing storage should be enough. https://github.com/onflow/cadence/blob/f5a7b6e3c84ca9757792018eb5a821f79b569cd5/runtime/storage.go#L230C18-L230C25 no need to set new storage

Reply:
Maybe the problem is a missing commit of the data?

Reply:
I will try it out :pray:

Reply:
@bluesign You mean this kind of context?
```go
ctx := runtime.Context{
	Interface:   r.backend.blockchain.NewScriptEnvironment(),
	Location:    testScriptLocation,
	Environment: env,
}
```

Reply:
I checked it out in detail, unfortunately not many options ( I am not even sure there is 100% correct solution ) 

Situation is `Test` is running in script context, but deployContract etc running transaction. One option for simple contracts can be syncing storage after deploy contract phase. ( basically getting Contract Composite from emulator storage, Clone(), and write to interpreter storage ) Even this passes current tests, unfortunately this doesn't work, if contract makes changes to somewhere else. ( for example creating another account, or saving something to account storage )

Reply:
@turbolent commit was my main thought too, but deploy is running a transaction, ofc we can always change deploy to deploy in the current context.

Reply:
I agree, had a look as well, and it is not straightforward at this point to 'share' the storage for the test-script and the tested code (deployed contract / transaction).

I think we can merge this as-is and unblock developers, and then look into a long-term solution separately.
Given this will be used only during tests, I don't think the overhead of copying would be a deal-breaker for now.

Reply:
Sounds good! Can we please create a follow-up issue (if it does not exist yet)?